hdeps/checkout.py

```diff
@@ -0,0 +1,39 @@
+import os.path
+from pathlib import Path
+
+from typing import List, Tuple
+
+from metadata_please import (
+    basic_metadata_from_pep621_checkout,
+    basic_metadata_from_setuptools_checkout,
+)
+
+from packaging.requirements import Requirement
+
+
+def read_checkout_reqs(path: Path) -> Tuple[str, List[Requirement]]:
+    try:
+        bm = basic_metadata_from_pep621_checkout(path)
+    except OSError:
+        pass
+    else:
+        if bm.reqs:
+            return (
+                os.path.join(path, "pyproject.toml"),
+                [Requirement(req) for req in bm.reqs],
+            )
+
+    try:
+        bm = basic_metadata_from_setuptools_checkout(path)
+    except OSError:
+        pass
+    else:
+        if bm.reqs:
+            return (
+                os.path.join(path, "setup.cfg"),
+                [Requirement(req) for req in bm.reqs],
+            )
```

Comment:
Do you anticipate expanding this to include `setup.py` or `requirements.in` or `requirements.txt`? 

Or is that out of scope?